src/Template/Deserialization.php

```diff
@@ -62,7 +62,16 @@ function {{functionName}}(array {{jsonPath}}): {{className}}
 EOT;
 
     private const TMPL_ASSIGN_DATETIME_FROM_FORMAT = <<<'EOT'
-{{modelPath}} = \DateTime::createFromFormat('{{format}}', {{jsonPath}});
+foreach([{{formats|join(', ')}}] as {{format}}) {
+    if (({{date}} = \DateTime::createFromFormat({{format}}, {{jsonPath}}, {{timezone}}))) {
+        {{modelPath}} = {{date}};
+        break;
+    }
+}
+
+if (false === ({{date}} ?? null)) {
```

Comment:
i think we should also fail if there have been no formats at all:

```suggestion
if (false === (date ?? false)) {
```

Reply:
done :+1: